pkg/azuredisk/controllerserver.go

```diff
@@ -1239,6 +1254,183 @@ func (d *Driver) ListSnapshots(ctx context.Context, req *csi.ListSnapshotsReques
 	return azureutils.GetEntriesAndNextToken(req, snapshots)
 }
 
+func (d *Driver) CreateVolumeGroupSnapshot(ctx context.Context, req *csi.CreateVolumeGroupSnapshotRequest) (*csi.CreateVolumeGroupSnapshotResponse, error) {
+	sourceVolumeIDs := req.GetSourceVolumeIds()
+	if len(sourceVolumeIDs) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "CreateVolumeGroupSnapshot Source Volume IDs must be provided")
+	}
+	for _, sourceVolumeID := range sourceVolumeIDs {
+		if len(sourceVolumeID) == 0 {
+			return nil, status.Error(codes.InvalidArgument, "CreateVolumeGroupSnapshot Source Volume ID must be provided")
+		}
+	}
+	volumeGroupSnapshotName := req.Name
+	if len(volumeGroupSnapshotName) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "volume group snapshot name must be provided")
+	}
+
+	snapshotNames := []string{}
+	for _, sourceVolumeID := range sourceVolumeIDs {
+		groupSnapshotUUID := strings.TrimPrefix(volumeGroupSnapshotName, "groupsnapshot-")
+		snapshotName := fmt.Sprintf("snapshot-%x", sha256.Sum256([]byte(groupSnapshotUUID+sourceVolumeID)))
```

Comment:
Here I use groupsnapshotuuid+sourceVolumeID to hash, since in snapshot-controller it use groupsnapshotcontentuuid+sourceVolumeID. But in CSI we can't get groupsnapshotcontent uuid

Reply:
@xing-yang we are generating snapshotName inside the `CreateVolumeGroupSnapshot`, do you think this is the right way now? does this snapshotName should be provided by `CreateVolumeGroupSnapshot` request?

Reply:
I think we should be able to use groupSnapshotUUID in snapshot-controller as well.

Reply:
https://github.com/kubernetes-csi/external-snapshotter/pull/1294 have a pr for this issue